cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/runtime/spi/runtimejob/TetheringRuntimeJobManager.java

```diff
@@ -132,4 +143,54 @@ void publishToControlChannel(TetheringControlMessage message) throws Exception {
       throw new Exception(String.format("Failed to publish to topic %s", topicId), e);
     }
   }
+
+  /**
+   * Use GZIPOutputStream to compress the LocalFile and Base64 to encode as String
+   */
+  @VisibleForTesting
+  String getLocalFileAsCompressedString(LocalFile localFile) throws IOException {
+    File file = new File(localFile.getURI());
+    byte[] buffer = new byte[1024 * 500]; // use 500kb buffer
+    ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    try (GZIPOutputStream os = new GZIPOutputStream(baos); FileInputStream fis = new FileInputStream(file)) {
+      int length;
+      while ((length = fis.read(buffer)) > 0) {
+        os.write(buffer, 0, length);
+      }
+    }
+    return Base64.encodeBase64String(baos.toByteArray());
```

Comment:
can we send the bytes directly? Rather than converting to string and then again to bytes for payload?